core/commands/add.go

```diff
@@ -270,7 +270,7 @@ You can now check what blocks have been created by:
 		fileAdder.Hidden = hidden
 		fileAdder.Trickle = trickle
 		fileAdder.Wrap = wrap
-		fileAdder.Pin = dopin
+		fileAdder.Pin = dopin && !hash
```

Comment:
And maybe we should do `fileAdder.Pin = dopin && !hash`,because `hash` is mean not to pin file.